frame/contracts/src/exec.rs

```diff
@@ -43,13 +43,82 @@ pub type StorageKey = [u8; 32];
 /// A type that represents a topic of an event. At the moment a hash is used.
 pub type TopicOf<T> = <T as frame_system::Config>::Hash;
 
-/// Describes whether we deal with a contract or a plain account.
-pub enum TransactorKind {
-	/// Transaction was initiated from a plain account. That can be either be through a
-	/// signed transaction or through RPC.
-	PlainAccount,
-	/// The call was initiated by a contract account.
-	Contract,
+/// Information needed for rent calculations that can be requested by a contract.
+#[derive(codec::Encode)]
+#[cfg_attr(test, derive(Debug, PartialEq))]
+pub struct RentParams<T: Config> {
+	/// The total balance of the contract. Includes the balance transferred from the caller.
```

Comment:
In ink! we have the possibility of instantiating contracts a la 
```
let contract = FooContract::new()
  .endowment(total_balance / 4)
  ...
  .instantiate()
  .expect("failed instantiating contract");
```
Do you have an idea what a succeeding `rent_params().total_balance` call would return for the parent contract in this case? I guess it should exclude the balance transferred to the sub-contract then.

Reply:
You mean the contract which executes the snippet you just posted? `rent_params()` is snapshotted at the beginning of execution of the parent contract. So it will return the same value before instantiating the child and after. But the contract knows how much it transferred away. So this should not be a problem.